src/dso_api/dynamic_api/permissions.py

```diff
@@ -21,6 +22,13 @@ class TableScopes:
 def fetch_scopes_for_dataset_table(dataset_id: str, table_id: str):
     """ Get the scopes for a dataset and table, based on the Amsterdam schema information """
 
+    # Make sure that names are snake_cased
+    # if used for a lookup in models.DatasetTable.objects
+    # TODO: do snake_case transformations at one centralized place to
+    # to be used in code (as much as possible)
+    dataset_id = to_snake_case(dataset_id)
```

Comment:
For now, this is a quick solution. As you already stated, we need to analyze the useage of the snake and camelcasing through the whole application (see issue 5207). The reason we need to do this the snakecasing here, is that the dataset name is snaked_cased before it it stored in the database (see DatasetTable model in schema-tools). Is that the right approach? These and other questions need to be addressed.

Reply:
I will post this context into the ticket, thnx